updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -141,76 +158,91 @@ def dependency_files_for_globs
       end.compact
     end
 
+    sig { returns(T::Array[DependencyFile]) }
     def dependency_files
-      return @dependency_files if defined?(@dependency_files)
+      return @dependency_files if defined?(@dependency_files) && !@dependency_files.nil?
 
-      @dependency_files = with_retries { file_fetcher.files }
+      @dependency_files = T.let(with_retries { file_fetcher.files }, T.nilable(T::Array[DependencyFile]))
       post_ecosystem_versions(file_fetcher) if should_record_ecosystem_versions?
-      @dependency_files
+      T.must(@dependency_files)
     end
 
+    sig { returns(T::Boolean) }
     def should_record_ecosystem_versions?
       # We don't set this flag in GHES because there's no point in recording versions since we can't access that data.
       Experiments.enabled?(:record_ecosystem_versions)
     end
 
+    sig { params(file_fetcher: FileFetchers::Base).void }
     def post_ecosystem_versions(file_fetcher)
       ecosystem_versions = file_fetcher.ecosystem_versions
       api_client.record_ecosystem_versions(ecosystem_versions) unless ecosystem_versions.nil?
     end
 
-    def with_retries(max_retries: 2)
+    sig { params(_blk: T.proc.returns(T.untyped)).returns(T.untyped) }
```

Comment:
Not quite what I meant. check out [the Sorbet documentation on generics][1]. I think it might be something like

```suggestion
    sig do
      type_parameters(:T)
      	.params(_blk: T.proc.returns(T.type_parameter(:T)))
      	.returns(T.type_parameter(:T)) }
    end
```

[1]: https://sorbet.org/docs/generics